plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeGcsConnectorSmokeTest.java

```diff
@@ -0,0 +1,294 @@
+                        .put("hive.metastore.thrift.client.max-retry-time", "1s")
+                        .buildOrThrow(),
+                dockerizedDataLake.getTestingHadoop(),
+                queryRunner -> {});
+    }
+
+    @Override
+    protected Optional<String> getHadoopBaseImage()
+    {
+        return Optional.of("ghcr.io/trinodb/testing/hdp3.1-hive");
+    }
+
+    @Override
+    protected void createTableFromResources(String table, String resourcePath, QueryRunner queryRunner)
+    {
+        String targetDirectory = bucketName + "/" + table;
+
+        try {
+            List<ClassPath.ResourceInfo> resources = ClassPath.from(TestDeltaLakeAdlsConnectorSmokeTest.class.getClassLoader())
+                    .getResources()
+                    .stream()
+                    .filter(resourceInfo -> resourceInfo.getResourceName().startsWith(resourcePath + "/"))
+                    .collect(toImmutableList());
+            for (ClassPath.ResourceInfo resourceInfo : resources) {
+                String fileName = resourceInfo.getResourceName().replaceFirst("^" + Pattern.quote(resourcePath), quoteReplacement(targetDirectory));
+                ByteSource byteSource = resourceInfo.asByteSource();
+                try (FSDataOutputStream fileStream = fileSystem.create(new org.apache.hadoop.fs.Path(fileName), true)) {
+                    ByteStreams.copy(byteSource.openBufferedStream(), fileStream);
+                }
+            }
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        queryRunner.execute(format("CREATE TABLE %s (dummy int) WITH (location = '%s')", table, getLocationForTable(bucketName, table)));
+    }
+
+    @Override
+    protected String getLocationForTable(String bucketName, String tableName)
+    {
+        return bucketUrl() + tableName;
+    }
+
+    @Override
+    protected List<String> getTableFiles(String tableName)
+    {
+        return listAllFilesRecursive(tableName);
+    }
+
+    @Override
+    protected List<String> listCheckpointFiles(String transactionLogDirectory)
+    {
+        return listAllFilesRecursive(transactionLogDirectory).stream()
+                .filter(path -> path.contains("checkpoint.parquet"))
+                .collect(toImmutableList());
+    }
+
+    private List<String> listAllFilesRecursive(String directory)
+    {
+        String path = bucketUrl() + directory;
+        ImmutableList.Builder<String> paths = ImmutableList.builder();
+
+        try {
+            RemoteIterator<LocatedFileStatus> files = fileSystem.listFiles(new org.apache.hadoop.fs.Path(path), true);
+            while (files.hasNext()) {
+                LocatedFileStatus file = files.next();
+                if (!file.isDirectory()) {
+                    paths.add(file.getPath().toString());
+                }
+            }
+        }
+        catch (FileNotFoundException e) {
+            return ImmutableList.of();
```

Comment:
why not fail in case that the directory is not present?

Reply:
There are tests that rely on this behavior. Directory `xyz` doesn't exist, and the test checks how many files are in `xyz`. Seems like a reasonable contract to me